test/e2e/dynamic_resource_e2e_test.go

```diff
@@ -3,103 +3,156 @@ package e2e
-		require.NoError(GinkgoT(), err)
-
-		gvr.Resource = "servicemonitors"
-		err = waitForGVR(dynamicClient, gvr, "my-servicemonitor", ns.GetName())
-		require.NoError(GinkgoT(), err)
+
+				catsrc, err = crc.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Create(catsrc)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Wait for the CatalogSource to be ready
+				catsrc, err = fetchCatalogOnStatus(crc, catsrc.GetName(), catsrc.GetNamespace(), catalogSourceRegistryPodSynced)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			AfterEach(func() {
+				// Delete CatalogSource
+				err = crc.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Delete(catsrc.GetName(), deleteOpts)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			Context("should install the operator successfully", func() {
+				It("creating a subscription to the CatalogSource", func() {
+
+					// Generate a Subscription
+					subName := genName("dynamic-resources")
+					cleanupSub := createSubscriptionForCatalog(crc, catsrc.GetNamespace(), subName, catsrc.GetName(), "etcd", "singlenamespace-alpha", "", v1alpha1.ApprovalAutomatic)
+
+					sub, err := fetchSubscription(crc, catsrc.GetNamespace(), subName, subscriptionHasInstallPlanChecker)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("should have created the expected prometheus objects", func() {
+
+						ipName := sub.Status.InstallPlanRef.Name
+						ip, err := waitForInstallPlan(crc, ipName, sub.GetNamespace(), buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseFailed, v1alpha1.InstallPlanPhaseComplete))
+						Expect(err).NotTo(HaveOccurred())
+
+						// Ensure the InstallPlan contains the steps resolved from the bundle image
+						expectedSteps := map[registry.ResourceKey]struct{}{
+							{Name: "my-prometheusrule", Kind: "PrometheusRule"}: {},
+							{Name: "my-servicemonitor", Kind: "ServiceMonitor"}: {},
+						}
+
+						// Verify Resource steps match expected steps
+						for _, step := range ip.Status.Plan {
+							key := registry.ResourceKey{
+								Name: step.Resource.Name,
+								Kind: step.Resource.Kind,
+							}
+							for expected := range expectedSteps {
+								if strings.HasPrefix(key.Name, expected.Name) && key.Kind == expected.Kind {
+									delete(expectedSteps, expected)
+								}
+							}
+						}
+						Expect(len(expectedSteps)).To(BeZero(), "Resource steps do not match expected: %#v", expectedSteps)
+
+						// Confirm that the expected types exist
+						gvr := schema.GroupVersionResource{
+							Group:    "monitoring.coreos.com",
+							Version:  "v1",
+							Resource: "prometheusrules",
+						}
+
+						err = waitForGVR(dynamicClient, gvr, "my-prometheusrule", ns.GetName())
+						Expect(err).NotTo(HaveOccurred())
+
+						gvr.Resource = "servicemonitors"
+						err = waitForGVR(dynamicClient, gvr, "my-servicemonitor", ns.GetName())
+						Expect(err).NotTo(HaveOccurred())
+					})
+
+					defer cleanupSub()
```

Comment:
Should this statement appear on the line after `cleanupSub` is assigned? If `fetchSubscription` returns an error, this will never be executed.